protocol-units/da-sequencer/node/src/block.rs

```diff
@@ -18,17 +23,71 @@ impl SequencerBlockDigest {
 	}
 }
 
-#[derive(Clone, Copy, Default, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
+#[derive(
+	Serialize, Deserialize, Clone, Copy, Default, Debug, PartialEq, Eq, Hash, PartialOrd, Ord,
+)]
 pub struct BlockHeight(pub u64);
 
-#[derive(Clone, Default, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
+#[derive(Serialize, Deserialize, Clone, Default, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
 pub struct SequencerBlock {
-	height: BlockHeight,
-	block: Block,
+	pub height: BlockHeight,
+	pub block: Block,
 }
 
 impl SequencerBlock {
+	/// Try to construct a SequencerBlock, but fail if it exceeds the max encoded size.
+	pub fn try_new(height: BlockHeight, block: Block) -> Result<Self, DaSequencerError> {
+		let sb = SequencerBlock { height, block };
+		sb.validate_size()?;
+		Ok(sb)
+	}
+
 	pub fn get_block_digest(&self) -> SequencerBlockDigest {
 		SequencerBlockDigest(*self.block.id().as_bytes())
 	}
+
+	pub fn validate_size(&self) -> Result<(), DaSequencerError> {
```

Comment:
Size validation should be done Tx after Tx. You get the Tx size, and you define if it can be added.

Reply:
Will be dealing with this in the `feat: da-seq produce_block` pr